### ISSUE TITLE
Fatal error in intellisense (master)

### ISSUE THREAD

**Issue by nosami** [mono/monodevelop#4092]
The IDE (master - 7.6.0.60) just crashed while I was typing.

```
FATAL ERROR [2018-03-08 14:22:00Z]: An unhandled exception has occured. Terminating Visual Studio? True
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.Ide.CodeCompletion.CompletionWindowManager.UpdateCursorPosition () [0x00043] in /Users/builder/data/lanes/432/0d559012/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/CompletionWindowManager.cs:192
  at MonoDevelop.Ide.Editor.Extension.CompletionTextEditorExtension.HandlePositionChanged (System.Object sender, System.EventArgs e) [0x00009] in /Users/builder/data/lanes/432/0d559012/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/CompletionTextEditorExtension.cs:744
  at (wrapper delegate-invoke) <Module>.invoke_void_object_EventArgs(object,System.EventArgs)
  at MonoDevelop.SourceEditor.SourceEditorView.OnCaretPositionChanged (System.EventArgs e) [0x00029] in /Users/builder/data/lanes/432/0d559012/source/monodevelop/main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs:2596
  at MonoDevelop.SourceEditor.SourceEditorView.HandlePositionChanged (System.Object sender, MonoDevelop.Ide.Editor.DocumentLocationEventArgs e) [0x00011] in /Users/builder/data/lanes/432/0d559012/source/monodevelop/main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs:362
```

> VS bug [#581449](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/581449)

**Comment by mrward** [comment]
Just had this with d15-7 build from Wrench - Visual Studio for Mac 7.5 build 837.


```
FATAL ERROR [2018-03-13 16:44:56Z]: An unhandled exception has occured. Terminating Visual Studio? True
System.NullReferenceException: Object reference not set to an instance of an object
  at MonoDevelop.Ide.CodeCompletion.CompletionWindowManager.UpdateCursorPosition () [0x00043] in /Users/builder/data/lanes/5965/b3876f43/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/CompletionWindowManager.cs:192 
  at MonoDevelop.Ide.Editor.Extension.CompletionTextEditorExtension.HandlePositionChanged (System.Object sender, System.EventArgs e) [0x00009] in /Users/builder/data/lanes/5965/b3876f43/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Extension/CompletionTextEditorExtension.cs:744 
  at (wrapper delegate-invoke) <Module>.invoke_void_object_EventArgs(object,System.EventArgs)
  at MonoDevelop.SourceEditor.SourceEditorView.OnCaretPositionChanged (System.EventArgs e) [0x00029] in /Users/builder/data/lanes/5965/b3876f43/source/monodevelop/main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs:2596 
  at MonoDevelop.SourceEditor.SourceEditorView.HandlePositionChanged (System.Object sender, MonoDevelop.Ide.Editor.DocumentLocationEventArgs e) [0x00011] in /Users/builder/data/lanes/5965/b3876f43/source/monodevelop/main/src/addins/MonoDevelop.SourceEditor2/MonoDevelop.SourceEditor/SourceEditorView.cs:362 
  at (wrapper delegate-invoke) System.EventHandler`1[MonoDevelop.Ide.Editor.CaretLocationEventArgs].invoke_void_object_TEventArgs(object,MonoDevelop.Ide.Editor.CaretLocationEventArgs)
  at MonoDevelop.Ide.Editor.Caret.OnPositionChanged (MonoDevelop.Ide.Editor.CaretLocationEventArgs args) [0x00008] in /Users/builder/data/lanes/5965/b3876f43/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/Caret.cs:63 
  at Mono.TextEditor.CaretImpl.OnPositionChanged (MonoDevelop.Ide.Editor.CaretLocationEventArgs args) [0x00016] in /Users/builder/data/lanes/5965/b3876f43/source/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/CaretImpl.cs:331 
  at Mono.TextEditor.CaretImpl.set_Location (MonoDevelop.Ide.Editor.DocumentLocation value) [0x00071] in /Users/builder/data/lanes/5965/b3876f43/source/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/CaretImpl.cs:96 
  at Mono.TextEditor.TextViewMargin.MousePressed (Mono.TextEditor.MarginMouseEventArgs args) [0x00482] in /Users/builder/data/lanes/5965/b3876f43/source/monodevelop/main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextViewMargin.cs:2369 
  at Mono.TextEditor.TextArea.OnButtonPressEvent (Gdk.EventButton e) [0x00099] in /Users/builder/data/lanes/5965/b3876f43/source/monodevelop/main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextArea.cs:1136 
  at Gtk.Widget.buttonpressevent_cb (System.IntPtr widget, System.IntPtr evnt) [0x00016] in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/gtk-sharp-None/gtk/generated/Widget.cs:508 
```

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.6.0.126 (master)

Pull Request #4152 merged by: Lluis Sanchez
Author: mono
Commit: 8bd50b38b9b683826608b4b78368bd65bf03685b (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.5.0.845 (d15-7)

Pull Request #4153 merged by: Lluis Sanchez
Author: mono
Commit: aaf59466ba99d80edd5c47689907989ac8796fbb (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #4152 merged by: Lluis Sanchez
Author: mono
Commit: 8bd50b38b9b683826608b4b78368bd65bf03685b (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-18] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #4152 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 8bd50b38b9b683826608b4b78368bd65bf03685b (xamarin/monodevelop)
